src/instance.js

```diff
@@ -970,12 +971,26 @@ Instance.prototype.setMetadata = function(metadata, gaxOptions, callback) {
     callback = gaxOptions;
     gaxOptions = {};
   }
+  var reqOpts = {
+    instance: extend({name: this.id}, metadata),
+    updateMask: {
+      paths: [],
+    },
+  };
+
+  const fieldsForMask = ['displayName', 'type', 'labels'];
+
+  fieldsForMask.forEach(field => {
+    if (reqOpts.instance[field]) {
```

Comment:
Hmm, setting the description to an empty string will probably fail since it's considered falsely. Looks like I did that in App Profile. Can you change that here? Test instead for  
`field in reqOpts.instance`